pkg/collector/corechecks/system/gpu/nvidia/jetson.go

```diff
@@ -0,0 +1,215 @@
+	tegraStatsPath string
+
+	// The command line options for tegrastats
+	commandOpts []string
+
+	metricsSenders []metricsSender
+}
+
+// regexFindStringSubmatchMap returns a map of strings where the keys are the name
+// of the submatch groups defined in its expressions and the values are the matches,
+// if any, of that group.
+// A return value of nil indicates no match.
+// The map will contain an empty key that holds the full match, if any. E.g.
+// map[""] = fullMatch
+func regexFindStringSubmatchMap(regex *regexp.Regexp, str string) map[string]string {
+	matches := regex.FindStringSubmatch(str)
+	if matches == nil {
+		return nil
+	}
+	result := make(map[string]string)
+	names := regex.SubexpNames()
+	for i, match := range matches {
+		result[names[i]] = match
+	}
+	return result
+}
+
+// getSizeMultiplier returns a multiplier for a given unit, i.e. kb = 1024, mb = 1024*1024 etc...
+// If the unit is not one of "kb", "mb" or "gb", it returns 1
+func getSizeMultiplier(unit string) float64 {
+	switch strings.ToLower(unit) {
+	case "kb":
+		return kb
+	case "mb":
+		return mb
+	case "gb":
+		return gb
+	default:
+		return 1
+	}
+}
+
+// Parses the output of tegrastats
+func (c *JetsonCheck) processTegraStatsOutput(tegraStatsOuptut string) error {
+	sender, err := aggregator.GetSender(c.ID())
+	if err != nil {
+		return err
+	}
+
+	for _, metricSender := range c.metricsSenders {
+		err = metricSender.SendMetrics(sender, tegraStatsOuptut)
+		if err != nil {
+			return err
+		}
+	}
+	sender.Commit()
+	return nil
+}
+
+// Run executes the check
+func (c *JetsonCheck) Run() error {
+	// Kill tegrastats it it runs for twice as long as the interval we specified, to avoid blocking
+	// the check forever
+	ctx, cancel := context.WithTimeout(context.Background(), 2*tegraStatsInterval*time.Millisecond)
+	defer cancel()
+	cmd := exec.CommandContext(ctx, c.tegraStatsPath, c.commandOpts...)
+
+	// Parse the standard output for the stats
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	go func() {
+		in := bufio.NewScanner(stdout)
+		if in.Scan() {
+			// We only need to read one line
+			line := in.Text()
+			log.Debugf("tegrastats: %s", line)
+			if err = c.processTegraStatsOutput(line); err != nil {
+				_ = log.Error(err)
+			}
+		} else {
+			_ = log.Warnf("tegrastats did not produce any output")
+		}
+		// Tegrastats keeps running forever, so kill it after trying to read
+		// one line of output
+		err = cmd.Process.Signal(os.Kill)
+		if err != nil {
+			_ = log.Errorf("unable to stop %s check: %s", checkName, err)
+		}
+	}()
+
+	// forward the standard error to the Agent logger
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+	go func() {
+		in := bufio.NewScanner(stderr)
+		for in.Scan() {
+			_ = log.Error(in.Text())
+		}
+	}()
+
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+
+	// No need to check the result since we kill the process, so err is normally != nil
+	_ = cmd.Wait()
```

Comment:
the coment `//nolint:errcheck` at the end of the line will disable the linter for that line.